### ISSUE TITLE
You can activate Mind Quickening Gem and The Restrained Essence of Sapphiron at the same time

### ISSUE THREAD

**Issue by EyeOfTheBeast** [slowtorta/turtlewow-bug-tracker#6436]
**Please, provide a clear description what the bug is:**
It's possible to activate the on-use effects of both the Mind Quickening Gem and The Restrained Essence of Sapphiron trinkets at the same time if you mash their keybinds hard enough simultaneously:

![trinketsbug](https://github.com/slowtorta/turtlewow-bug-tracker/assets/142637959/10cab425-4e58-4eba-9823-a25f11ed27be)

**Steps to reproduce the behavior with as much detail as possible:**
1.  Spam activate both of these trinkets at the same time. With luck, both of them will activate at the same time instead of putting the other one on cooldown.
2. This probably affects other on-use damage trinkets as well.

**IDs of affected creatures, items, quests or spells with a link to the relevant page:**
Mind Quickening Gem item: https://database.turtle-wow.org/?item=19339
The Restrained Essence of Sapphiron item: https://database.turtle-wow.org/?item=23046

**Expected behavior. Describe how it should work:**
It shouldn't be possible to activate the on-use effects of both of these trinkets at the same time in some cases. They should always put the other trinket on cooldown when activated, otherwise this would be too overpowered.

**Comment by twowTurtReynolds** [comment]
I can't replicate this. Is this something you and others are able to do consistently?

**Comment by Sringaram** [comment]
You just use a macro with trinket use calls. It worked like this in vanilla too.

**Comment by EyeOfTheBeast** [comment]
> I can't replicate this. Is this something you and others are able to do consistently?

I have done it a few times, it's just a bit rare to pull off.

**Comment by twowTurtReynolds** [comment]
> You just use a macro with trinket use calls. It worked like this in vanilla too.

In 2004-2006 retail? Or on other vmangos servers? If this is just a vmangos race condition bug we can fix, we should.

**Comment by Sringaram** [comment]
> > You just use a macro with trinket use calls. It worked like this in vanilla too.
> 
> In 2004-2006 retail? Or on other vmangos servers? If this is just a vmangos race condition bug we can fix, we should.

Both as far as I recall. Using ToeP + ZHC was a staple of vanilla PvP videos.

**Comment by EyeOfTheBeast** [comment]
> > > You just use a macro with trinket use calls. It worked like this in vanilla too.
> > 
> > 
> > In 2004-2006 retail? Or on other vmangos servers? If this is just a vmangos race condition bug we can fix, we should.
> 
> Both as far as I recall. Using ToeP + ZHC was a staple of vanilla PvP videos.

Yes this this was nerfed in Patch 1.10 of vanilla to no longer allow using 2 damage trinkets at once: https://www.engadget.com/2011-08-23-wow-archivist-patch-1-10-storms-of-azeroth.html#:~:text=Many%20trinkets%20will,the%20Devilsaur%20Eye.

So this shouldn't be possible on Turtle WoW, but it is in some cases if you get lucky and use both damage trinkets at the same time and the server lets you use both effects.

**Comment by Sringaram** [comment]
Yeah you can still do it in 1.12 clients by mashing a macro. 

I think a good way to solve this is to allow the use of dual trinkets but make effects of the same type exclusive (SP/AP) so only one can be active. It'd open up more options by letting you mix different kinds of actives while keeping the one active+one passive trinket as the superior PvE option in most scenarios.

**Comment by EyeOfTheBeast** [comment]
Video proof of the bug recorded just now: https://www.youtube.com/watch?v=gWRpNvPsGAY

**Comment by twowTurtReynolds** [comment]
Indeed it looks like something only possible via the lua api and not the blizz actionbars.

**Comment by balakethelock** [comment]
@twowTurtReynolds This should be tagged Core, & spell and mage tags removed (All classes can exploit this bug, seen warriors do it for example)

**Comment by ratkosrb** [comment]
https://github.com/slowtorta/turtlewow-bug-tracker/assets/35845488/e1f8d51a-fd6d-41c7-9adc-33b4776be728

I can't reproduce the issue. Even if i forcibly send packets to use both trinkets at the same time, it does not let me activate them both.
